colcon_ros/task/ament_cmake/build.py

```diff
@@ -60,10 +60,18 @@ async def build(self):  # noqa: D102
             skip_hook_creation=False,
             environment_callback=add_app_to_cpp)
 
+        # if the build has failed getting targets might not be possible
+        try:
+            has_install_target = await has_target(args.build_base, 'install')
+        except Exception:
+            if not rc:
```

Comment:
Would go with `rc == 0` myself, just for clarity's sake since it's not a boolean. But that's a style nit.

Reply:
`rc` might be `None` in case the build succeeded.

Reply:
Huh, per the docs I thought it was always numeric when the process had exited, and only `None` if still running:

https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process.returncode

Reply:
There are extensions which don't return a `Process.returncode` if there are no errors: e.g. https://github.com/colcon/colcon-gradle/blob/ea31e3c16bbf36d41d031777319a06da669f8aa6/colcon_gradle/task/gradle/build.py#L58-L84